src/main/java/org/jabref/gui/integrity/IntegrityCheckAction.java

```diff
@@ -40,8 +40,7 @@ public IntegrityCheckAction(JabRefFrame frame, StateManager stateManager, TaskEx
     public void execute() {
         BibDatabaseContext database = stateManager.getActiveDatabase().orElseThrow(() -> new NullPointerException("Database null"));
         IntegrityCheck check = new IntegrityCheck(database,
-                Globals.prefs.getFilePreferences(),
-                Globals.prefs.getCitationKeyPatternPreferences(),
+                Globals.prefs,
```

Comment:
You can pass the PreferencesService also in the constructor

Reply:
If you already inject prefs in the constructor, could you also try to get rid of JabRefFrame in the constructor? Maybe you have to inject DialogService and StateManager too in the constructor. But im not sure why the IntegrityCheckDialog needs the libraryTab... If its just for the UndoManager... well... that one could be injected too. 😁

Reply:
I just checked... that would be a major refactor. Not worth it in this PR.

Reply:
> You can pass the PreferencesService also in the constructor

I don't think I understand 😛 
Do you mean passing `preferencesService.getGeneralPreferences().getDefaultEncoding()` instead of merging them?
Or are you saying `Globals.prefs.getEntryEditorPreferences().shouldAllowIntegerEditionBibtex()` can be included merged as well?

Reply:
I am assuming the first one to decrease coupling.

Reply:
> that would be a major refactor. Not worth it in this PR.

Not sure I understand you. I think we can make it more specific than requiring a JabRef frame. We only need the current `LibraryTab.editEntryAndFocusField` but I am not sure if that changes all that much.

Reply:
Yes, it would be about loose coupling. I just think it's more important to merge a quick fix for a bug early, as well as to focus on one thing only in this pr and create a separate pr for the refactor, if that one would exceed the amount of lines changed too much.